src/main/java/com/elrond/erdkotlin/data/vm/VmRepositoryImpl.kt

```diff
@@ -2,14 +2,28 @@ package com.elrond.erdkotlin.data.vm
 
 import com.elrond.erdkotlin.data.api.ElrondProxy
 import com.elrond.erdkotlin.data.toDomain
-import com.elrond.erdkotlin.domain.vm.SmartContractQuery
-import com.elrond.erdkotlin.domain.vm.SmartContractOutput
 import com.elrond.erdkotlin.domain.vm.VmRepository
+import com.elrond.erdkotlin.domain.vm.query.QueryContractInput
+import com.elrond.erdkotlin.domain.vm.query.QueryContractOutput
+import com.elrond.erdkotlin.domain.vm.query.integer.QueryContractDigitOutput
+import com.elrond.erdkotlin.domain.vm.query.string.QueryContractStringOutput
 
 internal class VmRepositoryImpl(private val elrondProxy: ElrondProxy) : VmRepository {
 
-    override fun queryContract(smartContractQuery: SmartContractQuery): SmartContractOutput {
-        return requireNotNull(elrondProxy.queryContract(smartContractQuery).data).data.toDomain()
+    override fun queryContract(queryContractInput: QueryContractInput): QueryContractOutput {
+        return requireNotNull(elrondProxy.queryContract(queryContractInput).data).data.toDomain()
+    }
+
+    override fun queryContractHex(queryContractInput: QueryContractInput): QueryContractStringOutput {
```

Comment:
Why wouldn't extend the `queryContract` function and specify within a parameter the way to interpret the result? A call would look similar to `queryContract(input, ContractQueryEnum.Hex)`  and do the query to the endpoint specific to the enum value (as well as the return object formatting)

Reply:
It sounds good for the input part.

However, on the output side, I do not see how I could make it work.
The issue is the return type.
- `queryContract ` returns a complex object (containing may fields)
- `queryContractHex` and `queryContractString` both return a single `String`
- `queryContractInt` returns a single `BigInteger`

The only thing I can do is merging `queryContractHex`, `queryContractString` and `queryContractInt` in the repository (which is in the private part of the SDK) by returning a String.
Then I can format that string in the usecases
But I will have to keep separate usecases to differentiate the `String` and `BigInteger` cases.

I cannot merge `queryContract` with its siblings at all.

Reply:
that's right. :+1: